File: 2026-01/src/day_27/prms_138476/Solution.java
package day_27.prms_138476;

import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int t : tangerine) {
            map.put(t, map.getOrDefault(t, 0) + 1);
        }

        List<Integer> list = new ArrayList<>(map.values());
        list.sort(Comparator.reverseOrder());

        int sum = 0;
        int ans = 0;
        for (int v : list) {
            sum += v;
            ans++;

            if (sum >= k) break;
        }

        return ans;
    }
}

File: 2026-01/src/day_27/prms_138476/Solution.cpp
#include <algorithm>
#include <string>
#include <unordered_map>
#include <vector>

using namespace std;

int solution(int k, vector<int> tangerine) {
    unordered_map<int, int> mp;
    for (int t : tangerine) {
        mp[t]++;
    }

    vector<int> a;
    for (auto [k, v] : mp) {
        a.push_back(v);
    }
    sort(a.rbegin(), a.rend());

    int sum = 0;
    int ans = 0;
    for (int x : a) {
        sum += x;
        ans++;

        if (sum >= k) break;
    }

    return ans;
}
